File: AdventOfCode/2019/day1.py
f = open("./day1.txt")
data = f.readlines()
#print(sum([(int(d)//3)-2 for d in data]))
s = 0
while True:
    data = [(int(d)//3)-2 for d in data]
    data = [d for d in data if d>0]
    if len(data)==0:
        break
    s += sum(data)
print(s)

File: AdventOfCode/2020/1/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	file, err := os.Open("./day1.txt")
	if err!=nil {
		fmt.Println(err)
		return
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	scanner.Split(bufio.ScanLines)

	var nums = make(map[int]bool)

	for scanner.Scan() {
		num,err := strconv.Atoi(scanner.Text())
		if err!= nil {
			fmt.Println(err)
			return
		}
		if _, ok := nums[2020-num]; ok {
			fmt.Println(num*(2020-num))
		} 
		nums[num] = true
	}
}

// expected ans for the input is 646779